roles/tests-integration/tests/common/mod.rs

```diff
@@ -282,3 +286,86 @@ pub async fn start_template_provider(tp_port: u16) -> TemplateProvider {
     template_provider.generate_blocks(16);
     template_provider
 }
+
+pub async fn start_sv2_translator(upstream: SocketAddr) -> SocketAddr {
+    let upstream_address = upstream.ip().to_string();
+    let upstream_port = upstream.port();
+    let upstream_authority_pubkey = Secp256k1PublicKey::try_from(
+        "9auqWEzQDVyd2oe1JVGFLMLHZtCo2FFqZwtKA5gd9xbuEu7PH72".to_string(),
+    )
+    .expect("failed");
+    let listening_address = get_available_address();
+    let listening_port = listening_address.port();
+    let hashrate = measure_hashrate(3) as f32 / 100.0;
+    let min_individual_miner_hashrate = hashrate;
+    let shares_per_minute = 60.0;
+    let channel_diff_update_interval = 60;
+    let channel_nominal_hashrate = hashrate;
+    let downstream_difficulty_config =
+        translator_sv2::proxy_config::DownstreamDifficultyConfig::new(
+            min_individual_miner_hashrate,
+            shares_per_minute,
+            0,
+            0,
+        );
+    let upstream_difficulty_config = translator_sv2::proxy_config::UpstreamDifficultyConfig::new(
+        channel_diff_update_interval,
+        channel_nominal_hashrate,
+        0,
+        false,
+    );
+    let upstream_conf = translator_sv2::proxy_config::UpstreamConfig::new(
+        upstream_address,
+        upstream_port,
+        upstream_authority_pubkey,
+        upstream_difficulty_config,
+    );
+    let downstream_conf = translator_sv2::proxy_config::DownstreamConfig::new(
+        listening_address.ip().to_string(),
+        listening_port,
+        downstream_difficulty_config,
+    );
+
+    let config =
+        translator_sv2::proxy_config::ProxyConfig::new(upstream_conf, downstream_conf, 2, 2, 8);
+    let translator_v2 = translator_sv2::TranslatorSv2::new(config);
+    tokio::spawn(async move {
+        translator_v2.start().await;
+    });
+    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+    listening_address
+}
+
+fn measure_hashrate(duration_secs: u64) -> f64 {
+    let mut share = {
+        let mut rng = thread_rng();
+        let mut arr = [0u8; 80];
+        rng.fill(&mut arr[..]);
+        arr
+    };
+    let start_time = std::time::Instant::now();
+    let mut hashes: u64 = 0;
+    let duration = std::time::Duration::from_secs(duration_secs);
```

Comment:
I'm fine with this approach but it seems a bit convoluted and confusing to read/understand

I'm not convinced we really need to replicate the bitcoin protocol so closely if all we want is to know the capacity of the system to calculate sha256d (attention to the `d`, it's 2x sha256)

why not simply hash some simple input?

if you REALLY want to closely replicate what happens on bitcoin mining, I would suggest leveraging rust-bitcoin

you can initialize a [`Header`](https://docs.rs/bitcoin/latest/bitcoin/block/struct.Header.html) struct and then call `header.block_hash` to "mine" it.

that's how both `mining-device` and `sv1-mining-device` do it